statefun-flink/statefun-flink-core/src/main/java/org/apache/flink/statefun/flink/core/logger/UnboundedFeedbackLogger.java

```diff
@@ -99,11 +98,19 @@ private void flushToKeyedStateOutputStream() throws IOException {
     checkState(keyedStateOutputStream != null, "Trying to flush envelopes not in a logging state");
 
     final DataOutputView target = new DataOutputViewStreamWrapper(keyedStateOutputStream);
-    for (Entry<Integer, KeyGroupStream<T>> entry : keyGroupStreams.entrySet()) {
-      checkpointedStreamOperations.startNewKeyGroup(keyedStateOutputStream, entry.getKey());
+    final Iterable<Integer> assignedKeyGroupIds =
+        checkpointedStreamOperations.keyGroupList(keyedStateOutputStream);
+    // the underlying checkpointed raw stream, requires that all key groups assigned
+    // to this operator must be written to the underlying stream.
```

Comment:
I think that it would be better to write down the empty groups anyways, because they are presented to us on restore.
And having a header there, would help us to differentiate between different versions.
But I can definitely add a TODO: revist this after 19748 is merged. 
would that work for you?

Reply:
I don’t have a strong opinion on whether or not the empty key groups should stay there in the long term, so fine by me to keep this as is without the TODO comment to revisit 👍